osu.Game.Rulesets.Osu/Edit/Checks/CheckOffscreenObjects.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets.Edit.Checks.Components;
+using osu.Game.Rulesets.Osu.Objects;
+using osuTK;
+
+namespace osu.Game.Rulesets.Osu.Edit.Checks
+{
+    public class CheckOffscreenObjects : ICheck
+    {
+        // These are close approximates to the edges of the screen
+        // in gameplay on a 4:3 aspect ratio for osu!stable.
+        private const int min_x = -67;
+        private const int min_y = -60;
+        private const int max_x = 579;
+        private const int max_y = 428;
```

Comment:
Do you have a reference for how these were calculated? Probably something we want to get right from the start else it may be overlooked until it crops up in some mapping drama.

Reply:
Reference is a mix between cross-referencing with previous modding tools, my own testing using slider lengths in 1 px increments for every side of the screen, and 2 years of feedback from Mapset Verifier using these numbers (haven't had anyone report a false positive or negative for this check in a long time, and it's used by practically all BNs).

Only thing I'm unsure of is if the coords convert to lazer 1:1, but I suppose they would have to.

Reply:
So these are based on gameplay coordinate space, correct? (512x384 playfield / 640x480 screen area)